File: back-go/internal/shared/domain/business/gearanalysis.go
package business

type GearKind string

const (
	GearKindBike    GearKind = "BIKE"
	GearKindShoe    GearKind = "SHOE"
	GearKindUnknown GearKind = "UNKNOWN"
)

type GearAnalysis struct {
	Items      []GearAnalysisItem
	Unassigned GearAnalysisSummary
	Coverage   GearAnalysisCoverage
}

type GearAnalysisItem struct {
	ID                       string
	Name                     string
	Kind                     GearKind
	Retired                  bool
	Primary                  bool
	MaintenanceStatus        string
	MaintenanceLabel         string
	MaintenanceTasks         []GearMaintenanceTask
	MaintenanceHistory       []GearMaintenanceRecord
	Distance                 float64
	MovingTime               int
	ElevationGain            float64
	Activities               int
	AverageSpeed             float64
	FirstUsed                string
	LastUsed                 string
	LongestActivity          *ActivityShort
	BiggestElevationActivity *ActivityShort
	FastestActivity          *ActivityShort
	MonthlyDistance          []GearAnalysisPeriodPoint
}

type GearAnalysisSummary struct {
	Distance      float64
	MovingTime    int
	ElevationGain float64
	Activities    int
	AverageSpeed  float64
}

type GearAnalysisCoverage struct {
	TotalActivities      int
	AssignedActivities   int
	UnassignedActivities int
}

type GearAnalysisPeriodPoint struct {
	PeriodKey     string
	Value         float64
	ActivityCount int
}

type GearMaintenanceRecord struct {
	ID             string  `json:"id"`
	GearID         string  `json:"gearId"`
	GearName       string  `json:"gearName"`
	Component      string  `json:"component"`
	ComponentLabel string  `json:"componentLabel"`
	Operation      string  `json:"operation"`
	Date           string  `json:"date"`
	Distance       float64 `json:"distance"`
	Note           string  `json:"note,omitempty"`
	CreatedAt      string  `json:"createdAt"`
	UpdatedAt      string  `json:"updatedAt"`
}

type GearMaintenanceRecordRequest struct {
	GearID    string  `json:"gearId"`
	Component string  `json:"component"`
	Operation string  `json:"operation"`
	Date      string  `json:"date"`
	Distance  float64 `json:"distance"`
	Note      string  `json:"note,omitempty"`
}

type GearMaintenanceTask struct {
	Component         string                 `json:"component"`
	ComponentLabel    string                 `json:"componentLabel"`
	IntervalDistance  float64                `json:"intervalDistance"`
	IntervalMonths    int                    `json:"intervalMonths"`
	Status            string                 `json:"status"`
	StatusLabel       string                 `json:"statusLabel"`
	DistanceSince     float64                `json:"distanceSince"`
	DistanceRemaining float64                `json:"distanceRemaining"`
	NextDueDistance   float64                `json:"nextDueDistance"`
	MonthsSince       int                    `json:"monthsSince"`
	MonthsRemaining   int                    `json:"monthsRemaining"`
	LastMaintenance   *GearMaintenanceRecord `json:"lastMaintenance,omitempty"`
}

File: back-kotlin/src/main/kotlin/me/nicolas/stravastats/domain/business/GearAnalysis.kt
package me.nicolas.stravastats.domain.business

enum class GearKind {
    BIKE,
    SHOE,
    UNKNOWN,
}

data class GearAnalysis(
    val items: List<GearAnalysisItem>,
    val unassigned: GearAnalysisSummary,
    val coverage: GearAnalysisCoverage,
)

data class GearAnalysisItem(
    val id: String,
    val name: String,
    val kind: GearKind,
    val retired: Boolean,
    val primary: Boolean,
    val maintenanceStatus: String,
    val maintenanceLabel: String,
    val maintenanceTasks: List<GearMaintenanceTask> = emptyList(),
    val maintenanceHistory: List<GearMaintenanceRecord> = emptyList(),
    val distance: Double,
    val movingTime: Int,
    val elevationGain: Double,
    val activities: Int,
    val averageSpeed: Double,
    val firstUsed: String,
    val lastUsed: String,
    val longestActivity: ActivityShort?,
    val biggestElevationActivity: ActivityShort?,
    val fastestActivity: ActivityShort?,
    val monthlyDistance: List<GearAnalysisPeriodPoint>,
)

data class GearAnalysisSummary(
    val distance: Double,
    val movingTime: Int,
    val elevationGain: Double,
    val activities: Int,
    val averageSpeed: Double,
)

data class GearAnalysisCoverage(
    val totalActivities: Int,
    val assignedActivities: Int,
    val unassignedActivities: Int,
)

data class GearAnalysisPeriodPoint(
    val periodKey: String,
    val value: Double,
    val activityCount: Int,
)

data class GearMaintenanceRecord(
    val id: String,
    val gearId: String,
    val gearName: String,
    val component: String,
    val componentLabel: String,
    val operation: String,
    val date: String,
    val distance: Double,
    val note: String? = null,
    val createdAt: String,
    val updatedAt: String,
)

data class GearMaintenanceRecordRequest(
    val gearId: String = "",
    val component: String = "",
    val operation: String = "",
    val date: String = "",
    val distance: Double = 0.0,
    val note: String? = null,
)

data class GearMaintenanceTask(
    val component: String,
    val componentLabel: String,
    val intervalDistance: Double,
    val intervalMonths: Int,
    val status: String,
    val statusLabel: String,
    val distanceSince: Double,
    val distanceRemaining: Double,
    val nextDueDistance: Double,
    val monthsSince: Int,
    val monthsRemaining: Int,
    val lastMaintenance: GearMaintenanceRecord? = null,
)
